frontend/app/src/KeycloakDevTools.vue

```diff
@@ -0,0 +1,142 @@
+      >
+    </div>
+  </div>
+</template>
+<script>
+export default {
+  name: 'KeycloakDevTools',
+  data: function () {
+    return {
+      result: '',
+    };
+  },
+  methods: {
+    loadUserInfo: function () {
+      var vm = this;
+      this.$keycloak
+        .loadUserInfo()
+        .success(function (responseData) {
+          vm.result = JSON.stringify(responseData, null, '  ');
+        })
+        .error(function () {
+          vm.result = 'Failed to load user info';
+        });
+    },
+    loadProfile: function () {
+      var vm = this;
+      this.$keycloak
+        .loadUserProfile()
+        .success(function (profile) {
+          vm.result = JSON.stringify(profile, null, '  ');
+        })
+        .error(function () {
+          vm.result = 'Failed to load user info';
+        });
+    },
+    showToken: function () {
+      this.result = this.$keycloak.tokenParsed;
+    },
+    showRefreshToken: function () {
+      this.result = this.$keycloak.refreshTokenParsed;
+    },
+    showIdToken: function () {
+      this.result = this.$keycloak.idTokenParsed;
+    },
+    showDetails: function () {
+      this.result = this.$keycloak;
+    },
+    showExpires: function () {
+      if (!this.$keycloak.tokenParsed) {
+        this.result = 'Not authenticated';
+        return;
+      }
+      var o =
+        'Token Expires:\t\t' +
+        new Date(
+          (this.$keycloak.tokenParsed.exp + this.$keycloak.timeSkew) * 1000
+        ).toLocaleString() +
+        '\n';
+      o +=
+        'Token Expires in:\t' +
+        Math.round(
+          this.$keycloak.tokenParsed.exp +
+            this.$keycloak.timeSkew -
+            new Date().getTime() / 1000
+        ) +
+        ' seconds\n';
+
+      if (this.$keycloak.refreshTokenParsed) {
+        o +=
+          'Refresh Token Expires:\t' +
+          new Date(
+            (this.$keycloak.refreshTokenParsed.exp + this.$keycloak.timeSkew) *
+              1000
+          ).toLocaleString() +
+          '\n';
+        o +=
+          'Refresh Expires in:\t' +
+          Math.round(
+            this.$keycloak.refreshTokenParsed.exp +
+              this.$keycloak.timeSkew -
+              new Date().getTime() / 1000
+          ) +
+          ' seconds';
+      }
+
+      this.result = o;
+    },
+    updateProfile: function () {
+      var url = this.$keycloak.createAccountUrl().split('?')[0];
+      var req = new XMLHttpRequest();
+      req.open('POST', url, true);
+      req.setRequestHeader('Accept', 'application/json');
+      req.setRequestHeader('Content-Type', 'application/json');
+      req.setRequestHeader('Authorization', 'bearer ' + this.$keycloak.token);
+
+      var vm = this;
+      req.onreadystatechange = function () {
+        if (req.readyState === 4) {
+          if (req.status === 200) {
+            vm.result = 'Success';
+          } else {
+            vm.result = 'Failed';
+          }
+        }
+      };
+
+      req.send(
```

Comment:
I'm wondering about the purpose the line. Does it send email?

Reply:
Basically this function updateProfile is used to Update profile but unfortunately this function never succeeded due to permissions. req.send(PARAMS) would be send post request to the created URL above to update profile. As I mentioned this request never get satisfied due to permissions, Moreover this action can be performed on local environment only the whole page would be shown on local only for testing purpose. Here is screenshot:
<img width="1510" alt="Screenshot 2024-03-04 at 4 08 36 PM" src="https://github.com/bcgov/moh-phlat/assets/107434397/55fc5fb7-a7df-4273-87b4-2fcfee75cae0">